cds_ils/importer/views.py

```diff
@@ -76,15 +82,28 @@ def __init__(self, serializers=None, *args, **kwargs):
     @need_permissions("document-importer")
     def get(self, log_id, offset):
         """Returns the detail views of each subtask by given offset."""
-        log = None
         try:
             log = db.session.query(ImporterImportLog).get(log_id)
-        except ObjectDeletedError:
-            abort(404)
-        if log:
             return self.make_response(log, record_offset=offset)
-        else:
-            abort(404, "Task not found")
+        except ObjectDeletedError:
+            abort(404, "The task log was deleted.")
+        except NoResultFound:
+            abort(404, "Task does not exist.")
+
+    def post(self, log_id):
```

Comment:
I think that you might have forgotten the permission check?
`@need_permissions("document-importer")`